src/python/zquantum/core/estimation/_estimation.py

```diff
@@ -219,6 +219,57 @@ def evaluate_estimation_circuits(
     ]
 
 
+def get_estimation_tasks_to_measure(
```

Comment:
I think the way you do it is ok, but I think we could improve it slightly:
1. Take all the estimation tasks
2. Split it into two groups: executable and non-executable
3. Then treat "executable" as usual and "non-executable" separately
4. Combine the results at the end.

I think it still would require keeping track of the indices in the expectation values, but at least it would more explicit.
Also, `get_estimation_tasks_to_measure` actually does two things – it gives you the estimation tasks you want to use and it evaluates those with constant terms. And therefore it breaks [single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle). 

 (which might be cleaner and more in line with the conventions we follow).

Reply:
I think your suggestion makes a lot of sense ! I'm thinking that I will then split `get_estimation_tasks_to_measure` into two functions:
- one that will split the EstimationTasks into two groups
- one that evaluates EstimationTasks that have constant terms

Does that seem like a good plan to you?

Reply:
Yes, sounds like plan!